### ISSUE TITLE
[question] How to change the icon of an admonition?

### ISSUE THREAD

**Issue by 2bndy5** [jbms/sphinx-immaterial#5]
My intention is to have 
![image](https://user-images.githubusercontent.com/14963867/136675648-5a51c27b-b07f-4921-8a7a-06f9841b7a4c.png)
look like
![image](https://user-images.githubusercontent.com/14963867/136675659-63bce8d4-418c-46d8-83c6-538c1557a4fc.png)

But it looks like the built theme only allows a limited set of options. So, how do I change the icon for admonitions? Usually I'd override the CSS styling to do this, but I don't see the compiled svg I want.

**Comment by 2bndy5** [comment]
## Solution
This info should go in the revised sphinx-immaterial docs since the theme's assets are programmatically optimized before distribution.

1. Find the desired icon from the https://materialdesignicons.com/ listing
2. copy the "Inline SVG for HTML" code
3. paste it into 
   ```css
   .md-typeset .admonition.tip .admonition-title::before {
     mask-image: url('data:image/svg+xml;charset=utf-8,{copied code goes here}'); /* no the curly brackets */
     background-color: hsl(301, 100%, 63%); /* to change the color of the icon (if desired) */
   }
   ```

This method does not work well with nested admonitions, but I may be overriding the CSS poorly.

**Comment by 2bndy5** [comment]
Well. I found the icons listed on https://materialdesignicons.com/, but they have style data embedded in the SVG code.
I was able to get it working by copying the SVG code from the site and then
1. replace the `style="width:24px;height:24px" ` with `xmlns="http://www.w3.org/2000/svg" ` in the beginning `<svg>` tag
2. remove the `fill="currentColor" ` from the `<path>` tag's content
3. put altered svg code into a `mask-image` attribute for the css rule

Results:
```css
.md-typeset .admonition.tip .admonition-title::before,
.md-typeset .admonition.hint .admonition-title::before {
  mask-image: url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12,3L1,9L12,15L21,10.09V17H23V9M5,13.18V17.18L12,21L19,17.18V13.18L12,17L5,13.18Z" /></svg>');
}
```

**Comment by 2bndy5** [comment]
In case anyone has trouble with this solution after updating to sphinx-immaterial v0.3, the CSS rules have changed a bit. The following can be used to change the admonition's color and title icon:
```css
/* changing the style for the `.. seealso::` admonition */

.md-typeset .admonition.seealso>.admonition-title::before {
  mask-image: url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 576 512"><path d="M572.52 241.4C518.29 135.59 410.93 64 288 64S57.68 135.64 3.48 241.41a32.35 32.35 0 0 0 0 29.19C57.71 376.41 165.07 448 288 448s230.32-71.64 284.52-177.41a32.35 32.35 0 0 0 0-29.19zM288 400a144 144 0 1 1 144-144 143.93 143.93 0 0 1-144 144zm0-240a95.31 95.31 0 0 0-25.31 3.79 47.85 47.85 0 0 1-66.9 66.9A95.78 95.78 0 1 0 288 160z"/></svg>');
  background-color: hsl(301, 100%, 63%);
}

.md-typeset :is(.admonition):is(.seealso) {
  border-color: hsl(301, 100%, 63%);
}

.md-typeset :is(.seealso) > :is(.admonition-title) {
  background-color: hsla(287, 100%, 63%, 0.25);
  border-color: hsl(301, 100%, 63%);
}
```

**Comment by jbms** [comment]
I think it would be great if this could be customized from Python (and we could avoid having to inline svgs).

**Comment by 2bndy5** [comment]
I'm waiting on sphinx-design to publish a new release because I submitted a set a of material icons (based on Google's material icons repo) that can be used as inline text (it is currently merged into master). When that feature goes live, we could just rely on sphinx-design more.

**Comment by jbms** [comment]
The existing build system copies all of the icons in svg format into the sphinx-immaterial package, so from Python code we could generate appropriate CSS that inlines them.

**Comment by 2bndy5** [comment]
Are you proposing an optional CSS stylesheet named ("material_icons.sha.min.css") be enabled via a conf.py option called `export_svg_css = True`?

**Comment by 2bndy5** [comment]
In submitting the feature to sphinx-design, I used a python script that reads SVG data as XML. Since sphinx-design implements the icons via RST roles, it only pulls in what it needs (from a compiled JSON that I submitted) during `sphinx-build`.

I suppose we could extend this idea to FA icons and octicons also.

**Comment by 2bndy5** [comment]
How would we namespace the CSS vars? Would it be based on the theme assets that they're copied to?

```css
:root {
  /* sphinx_immaterial/.icons/material/ab-testing.svg file as a CSS var */
  --material__ab-testing: /* minified SVG data */;
}
```

**Comment by jbms** [comment]
What I had in mind was specifically focused on custom admonitions, to define some python functions (or perhaps it could be exposed via sphinx directives) like:

```python

def define_custom_admonition(name: str, default_title: str, icon_path: str, color: str): ...
```

Or something like that.

And that would arrange to define a new sphinx directive named `name` for the admonition type and also ensure appropriate CSS gets included.

**Comment by nheinsdorf** [comment]
Hey thanks for the super helpful thread! 
I also want to change the logo of an admonition. Putting the code snippet provided above into my custom.css file works for me, but if I want to use a different icon from the materials font, let's say the star. Where do I get the link from? 
I guess I have to replace this line: 
`

`mask-image: url('data:image/svg+xml;charset=utf-8,<svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 576 512"><path d="M572.52 241.4C518.29 135.59 410.93 64 288 64S57.68 135.64 3.48 241.41a32.35 32.35 0 0 0 0 29.19C57.71 376.41 165.07 448 288 448s230.32-71.64 284.52-177.41a32.35 32.35 0 0 0 0-29.19zM288 400a144 144 0 1 1 144-144 143.93 143.93 0 0 1-144 144zm0-240a95.31 95.31 0 0 0-25.31 3.79 47.85 47.85 0 0 1-66.9 66.9A95.78 95.78 0 1 0 288 160z"/></svg>');
`
thanks!

**Comment by jbms** [comment]
The information in this thread is not up to date --- refer to the documentation here:
https://jbms.github.io/sphinx-immaterial/admonitions.html#custom-admonitions

It is not longer necessary to write custom css at all in order to accomplish this.

**Comment by nheinsdorf** [comment]
Thank you so much for the answer! If I wanted to put it in the css file though, e.g. using a different theme. Where could I get the link that is provided in the example above? 
I'm using the 'material' theme, since immaterial doesn't work for me, I cannot install the google fonts...

**Comment by 2bndy5** [comment]
You should really refer to the docs for which theme you are using.

The SVG data exemplified here (and [in the mkdocs-material theme's docs](https://squidfunk.github.io/mkdocs-material/reference/admonitions/#custom-admonitions)) was extracted directly from [upstream's material/.icons/fontawesome/solid/brands/pied-piper-alt.svg](https://github.com/squidfunk/mkdocs-material/blob/master/material/.icons/fontawesome/brands/pied-piper-alt.svg). If you have the theme installed, then you can browse the install path to look for the SVG data; this requires an understanding of how/where python packages are installed though. Alternatively, you could browse the respective icon sources' sites ([listed in upstream's docs here](https://squidfunk.github.io/mkdocs-material/reference/icons-emojis/#configuration)), but some fontawesome icons may not be included with theme distribution due to licensing concerns.

---
> I cannot install the google fonts...

Yeah, we've been discussing a solution to this but haven't realized a solution yet.
